### ISSUE TITLE
ClearML callback uses local output path `project` as ClearML `project_name`, causing failures with absolute paths

### ISSUE THREAD

**Issue by xx025** [ultralytics/ultralytics#24820]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Ultralytics currently passes `trainer.args.project` directly to ClearML as `project_name`:

```python
task = Task.init(
    project_name=trainer.args.project or "Ultralytics",
    task_name=trainer.args.name,
    ...
)
```

This can cause problems because in Ultralytics, `project` is commonly used as a local filesystem output directory, for example:

```python
model.train(
    project="/path/to/output/runs/experiment_group",
    name="trial_0",
)
```

However, ClearML interprets `project_name` as a ClearML project name/path, not as a filesystem output path.

When `project` is an absolute path starting with `/`, the value is passed directly to ClearML as the ClearML project name. In my environment, this triggered a ClearML project lookup/create mismatch.

The lookup used the project name with a leading slash:

```text
name=/tmp/test/runs/optuna_202606012
```

but the existing project reported by ClearML had the leading slash removed:

```text
name=tmp/test/runs/optuna_202606012
```

This caused later task creation to fail with:

```text
Action failed <400/801: projects.create/v1.0 (Project with the same name already exists: name=tmp/test/runs/optuna_202606012, company=<COMPANY_ID>)> (name=/tmp/test/runs/optuna_202606012, description=, method=get)
```

### Minimal ClearML reproduction

This reproduces the underlying ClearML behavior:

```python
from clearml import Task


def main():
    for i in range(2):
        task = Task.init(
            project_name="/tmp/test/runs/optuna_202606012",
            task_name=f"repro_{i}",
            reuse_last_task_id=False,
        )
        task.close()


if __name__ == "__main__":
    main()
```

Removing the leading slash avoids the issue:

```python
project_name="tmp/test/runs/optuna_202606012"
```

### Why this affects Ultralytics

Ultralytics maps the training arguments to ClearML like this:

```python
ClearML project_name = trainer.args.project
ClearML task_name = trainer.args.name
```

So when users set `project` to an absolute output directory, the absolute filesystem path is sent to ClearML as the ClearML project name.

### Expected behavior

Ultralytics should avoid passing a local filesystem path directly as ClearML `project_name`, or it should sanitize the value before passing it to ClearML.

Possible minimal fix:

```python
project_name=str(trainer.args.project or "Ultralytics").lstrip("/")
```

This would separate the local training output directory from the ClearML project namespace.

### Affected versions

Confirmed in:

- `ultralytics==8.4.0`
- `ultralytics==8.4.66`

The current `8.4.66` ClearML callback still contains:

```python
project_name=trainer.args.project or "Ultralytics"
```

### Environment

- Ultralytics: `8.4.0`
- Also checked latest PyPI release: `8.4.66`
- ClearML Server: self-hosted Docker
- ClearML Server image: `clearml/server:latest`
- ClearML package inside server image: `2.1.3`
- Python: `3.10`

---

Ultralytics 8.4.0 🚀 Python-3.10.19 torch-2.10.0.dev20251208+cu128 CUDA:0 (NVIDIA GeForce RTX 3070, 7841MiB)
Setup complete ✅ (24 CPUs, 62.7 GB RAM, 47.1/231.2 GB disk)

OS                     Linux-5.15.0-174-generic-x86_64-with-glibc2.35
Environment            Linux
Python                 3.10.19
Install                pip
Path                    /conda/envs/yolo/lib/python3.10/site-packages/ultralytics
RAM                    62.67 GB
Disk                   47.1/231.2 GB
CPU                    AMD Ryzen 7 8700G w/ Radeon 780M Graphics
CPU count              24
GPU                    NVIDIA GeForce RTX 3070, 7841MiB
GPU count              2
CUDA                   12.8

numpy                  ✅ 2.2.6>=1.23.0
matplotlib             ✅ 3.10.8>=3.3.0
opencv-python          ✅ 4.12.0.88>=4.6.0
pillow                 ✅ 12.0.0>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.32.5>=2.23.0
scipy                  ✅ 1.15.3>=1.4.1
torch                  ✅ 2.10.0.dev20251208+cu128>=1.8.0
torch                  ✅ 2.10.0.dev20251208+cu128!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.25.0.dev20251208+cu128>=0.9.0
psutil                 ✅ 7.1.3>=5.8.0
polars                 ✅ 1.36.0>=0.20.0
ultralytics-thop       ✅ 2.0.18>=2.0.18

### Minimal Reproducible Example

```



from ultralytics import YOLO


def main():
    for i in range(2):
        model = YOLO("yolo11n.pt")
        model.train(
            data="coco8.yaml",
            epochs=1,
            imgsz=640,
            batch=8,
            project="/tmp/test/runs/optuna_202606012",
            name=f"repro_{i}",
            exist_ok=True,
            save=False,
            plots=False,
            val=False,
        )


if __name__ == "__main__":
    main()
````

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @xx025, thank you for the detailed bug report and for providing a clear reproducible example 🚀 This is an automated response to help triage your issue, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.

Since this is a 🐛 Bug Report, thanks for already including a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) and detailed environment information. This is very helpful for debugging ClearML integration issues involving `project` handling and callback behavior.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by xx025** [comment]
https://github.com/clearml/clearml/issues/1643

**Comment by xx025** [comment]
This is primarily a ClearML normalization bug. The same project_name value is not handled consistently across project creation and project lookup. Ultralytics only exposes the issue because it commonly passes an absolute filesystem path as project_name.

**Comment by Y-T-G** [comment]
You can use `save_dir` instead of `project` to specify the output directory. It's the more correct and recommended approach.

**Comment by Y-T-G** [comment]
@raimbekovm Can you document `save_dir` in training args?
